zilliqa/tests/it/main.rs

```diff
@@ -821,15 +815,11 @@ impl Network {
             .unwrap()
     }
 
-    pub async fn genesis_wallet(
-        &mut self,
-    ) -> SignerMiddleware<Provider<LocalRpcClient>, LocalWallet> {
+    pub async fn genesis_wallet(&mut self) -> Wallet {
```

Comment:
Does the genesis wallet have the same semantics as the coinbase wallet on ZQ1 i.e. "holding" all not yet minted native tokens and receiving burned txn fees? On ZQ1 the coinbase address is 0x0, but according to the wallet creation in the next line, x-shards will have a non-zero genesis wallet address, right?

Reply:
The genesis wallets are pre-funded in the initial state according to the config passed to the genesis nodes (see [https://github.com/Zilliqa/zq2/blob/766617811202715b9b3d8ca9d8d15be82c60ee5e/zilliqa/src/state.rs#L73](src/state.rs)). There may be more than one such wallet. Other than minting some tokens to their balance, they are not special in any way.
Currently their main usecase is as easy pre-funded wallets to use during testing.

I'm not sure how we'll handle minting on mainnet - possibly it'll be allocated to genesis wallets, but e.g. validator fees can be minted on the go and don't have to be "held" by the genesis wallet.

Reply:
Regarding mainnet ZIL supply we had a short discussion in whitepaper comments: https://docs.google.com/document/d/1wkYv7GV_-2X9Ha7Di56vjcNhS_H1o6n5rFbKz4xviYQ/edit?disco=AAABD6HQ3pQ

I guess we will import the null account from the ZQ1 persistence and from that wallet we will mint the mainnet validator rewards.